t/lib/OpenQA/Test/Utils.pm

```diff
@@ -42,14 +44,37 @@ BEGIN {
 
 our (@EXPORT, @EXPORT_OK);
 @EXPORT_OK = (
+    qw(mock_service_ports),
     qw(redirect_output standard_worker create_user_for_workers),
     qw(create_webapi create_websocket_server create_scheduler create_live_view_handler),
-    qw(unresponsive_worker broken_worker rejective_worker wait_for_worker setup_share_dir setup_fullstack_temp_dir run_gru_job),
+    qw(unresponsive_worker broken_worker rejective_worker setup_share_dir setup_fullstack_temp_dir run_gru_job),
     qw(collect_coverage_of_gru_jobs stop_service start_worker unstable_worker fake_asset_server),
     qw(cache_minion_worker cache_worker_service shared_hash embed_server_for_testing),
     qw(run_cmd test_cmd)
 );
 
+# The function OpenQA::Utils::service_port method hardcodes ports in a
+# sequential range starting with OPENQA_BASE_PORT. This can cause problems
+# especially in repeated testing if any of the port in that range is already
+# occupied, so we inject random, free ports for the services here
```

Comment:
Better end the sentence after "occupied". But nicely written comment (for code which otherwise may look odd).